### DISCUSSION TITLE
Regions pointer events when drag and resize is false

### DISCUSSION THREAD

**Discussion by mmilad** [katspaugh/wavesurfer.js#3475] [Q&A] [answered]
Hi,

I have a case where i add regions with drag and resize set to false, so i wondered if it wouldn´t be correct to assume that the region with both flags set to false would have its pointer events set to none.

luckily the addRegion method returns an object with the region element attached to it, so i can set the pointerEvents myself, but im not sure if this is a hack or a correct way to do this.

and thank you for this grad library!

**Comment by katspaugh** [comment, marked as answer]
The events are attached regardless of the resize/drag params, but inside the handler it checks and does nothing if those actions are disabled.

It's done in this way to avoid subscribing and unsubscribing when you change these options dynamically via setOptions.

I'm open to changing this behavior tho, please feel free to make a PR to attach drag events conditionally. How does it affect your usage btw?

**Comment by mmilad** [comment]
Good point! removing the pointer events would affect custom event registrations. 

Sorry i didnt really mention what i wanted to achive.
the regions catch the mouse events, even though they are only meant to be decorative, so i can´t click or drag (seek) the waveform at their positions.

my usage isn´t much affected because i get the element reference from the addRegion method and can set the css property myself, but it was just a bit unintuitive, as i would expect to have a flag or a style object that i could set, but i think that is only because i use the regions plugin as markers without any functionality.

Thank you!

**Reply by katspaugh** [reply]
True, it would be good to be able to drag “under” an undraggable region.

Clicks should work regardless though.
